x/approvals/context.go

```diff
@@ -0,0 +1,128 @@
+package approvals
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/x"
+)
+
+type contextKey int // local to the multisig module
+
+const (
+	contextKeyApprovals contextKey = iota
+	NoCount                        = -1
+	NoTimeout                      = -1
+)
+
+// withApproval is a private method, as only this module
+// can add a approved signer
+func withApproval(ctx weave.Context, id []byte) weave.Context {
+	val, _ := ctx.Value(contextKeyApprovals).([]weave.Condition)
+	if val == nil {
+		return context.WithValue(ctx, contextKeyApprovals, []weave.Condition{ApprovalCondition(id, "usage")})
+	}
+
+	return context.WithValue(ctx, contextKeyApprovals, append(val, ApprovalCondition(id, "usage")))
+}
+
+// ApprovalCondition returns condition for a given action and signer
+func ApprovalCondition(id []byte, approvalType string) weave.Condition {
+	return weave.NewCondition("approval", approvalType, id)
+}
+
+func ApprovalConditionWithCount(id []byte, approvalType string, count int64) weave.Condition {
```

Comment:
Okay, this is getting ugly.
You know introduce lots of info separated by `:` in the last field.
However, `id` may include `:` as well (it is random binary data, and I had regexp tests fail when an address happened to include a newline).